src/execution_plan/optimizations/reduce_scans.c

```diff
@@ -7,8 +7,19 @@
 #include "./reduce_scans.h"
 #include "../../util/arr.h"
 #include "../ops/op_conditional_traverse.h"
+#include "../ops/op_node_by_label_scan.h"
+#include "../ops/op_filter.h"
 #include <assert.h>
 
+static FT_FilterNode *buildLabelFilter(NodeByLabelScan *label_scan) {
+	AR_ExpNode *const_label_node = AR_EXP_NewConstOperandNode(SI_ConstStringVal((
+																					char *)label_scan->n->label));
```

Comment:
Can you rewrite a bit to make this call look more normal?

Reply:
yep. done